src/WebJobs.Extensions.OpenAI.CosmosDBNoSqlSearch/CosmosDBNoSqlSearchProvider.cs

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.WebJobs.Extensions.OpenAI.Search;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Extensions.OpenAI.CosmosDBNoSqlSearch;
+
+sealed class CosmosDBSearchProvider : ISearchProvider
+{
+    readonly IConfiguration configuration;
+    readonly ILogger logger;
+    readonly IOptions<CosmosDBNoSqlSearchConfigOptions> cosmosDBNoSqlSearchConfigOptions;
+    readonly ConcurrentDictionary<string, CosmosClient> cosmosDBClients = new();
+    string databaseName = "openai-functions-database";
+    string collectionName = "openai-functions-collection";
+
+    public string Name { get; set; } = "CosmosDBNoSqlSearch";
+
+    /// <summary>
+    /// Initializes CosmosDB search provider.
+    /// </summary>
+    /// <param name="configuration">The configuration.</param>
+    /// <param name="loggerFactory">The logger factory.</param>
+    /// <param name="cosmosDBNoSqlSearchConfigOptions">Cosmos DB No Sql search config options.</param>
+    /// <exception cref="ArgumentNullException">Throws ArgumentNullException if logger factory is null.</exception>
+    public CosmosDBSearchProvider(
+        IConfiguration configuration,
+        ILoggerFactory loggerFactory,
+        IOptions<CosmosDBNoSqlSearchConfigOptions> cosmosDBNoSqlSearchConfigOptions
+    )
+    {
+        this.configuration =
+            configuration ?? throw new ArgumentNullException(nameof(configuration));
+        if (loggerFactory == null)
+        {
+            throw new ArgumentNullException(nameof(loggerFactory));
+        }
+        this.cosmosDBNoSqlSearchConfigOptions = cosmosDBNoSqlSearchConfigOptions;
+    }
+
+    /// <summary>
+    /// Add a document to the search index.
+    /// </summary>
+    /// <param name="document">The searchable document.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>Returns a task that completes when the document is successfully saved.</returns>
+    public async Task AddDocumentAsync(
+        SearchableDocument document,
+        CancellationToken cancellationToken
+    )
+    {
+        CosmosClient cosmosClient = cosmosDBClients.GetOrAdd(
+            document.ConnectionInfo!.ConnectionName,
+            _ => CreateCosmosClient(document.ConnectionInfo!)
+        );
+
+        var databaseProperties = this.cosmosDBNoSqlSearchConfigOptions.Value.DatabaseProperties;
```

Comment:
Use the explicit type instead of var

(comment applicable for other occurrences too)